src/cmd/descriptor.rs

```diff
@@ -64,23 +62,25 @@ pub struct CountEdges {
 
 impl CountEdges {
     pub fn run(&self) -> Result<()> {
-        let descriptor = Descriptor::from_json(&self.input)
+        let descriptor = Descriptor::from_path(&self.input)
             .context(format!("reading descriptor {}", self.input.display()))?;
-        let mut counts: HashMap<&PublicKeyBinary, i32> = HashMap::new();
+        let mut counts: HashMap<&Vec<u8>, i32> = HashMap::new();
```

Comment:
I don't think `serde_json` would write `Vec<u8>` to the `edgecount.json` file. You'd hit `Error: key must be a string`. I think it's probably okay to leave the key for counts map as a `PublicKeyBinary` since it can be serialized to a string by serde.